kratos/processes/compute_nodal_normal_divergence_process.cpp

```diff
@@ -31,8 +31,22 @@ void ComputeNodalNormalDivergenceProcess<THistorical>::Execute()
     {
         Matrix J0, InvJ0, DN_DX;
     };
+    TLSType tls;
 
     // First element iterator
+    const auto it_element_begin = mrModelPart.ElementsBegin();
+
+    // Current domain size
+    const std::size_t dimension = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+
+    // Initial resize
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t number_of_nodes_first_element = r_first_element_geometry.PointsNumber();
+    const std::size_t local_space_dimension_first_element = r_first_element_geometry.LocalSpaceDimension();
+    if (tls.DN_DX.size1() != number_of_nodes_first_element || tls.DN_DX.size2() != dimension)
+        tls.DN_DX.resize(number_of_nodes_first_element, dimension);
+    if (tls.J0.size1() != dimension || tls.J0.size2() != local_space_dimension_first_element)
+        tls.J0.resize(dimension, local_space_dimension_first_element);
```

Comment:
:+1: Makes sense. Indeed I think this is more efficient since otherwise the TLS needs to check and resize for the first usage in each thread.

Just for my curiosity, didn't work before?

Reply:
It works fine without prior resizing (will be checked in the functions that calculate J0 and DN_DX). I did the change only for the sake of efficiency.

Reply:
OK. I just wanted to confirm that. Thanks!